vault/resource_kubernetes_secret_backend.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/hashicorp/terraform-provider-vault/util"
+)
+
+const (
+	fieldKubernetesHost    = "kubernetes_host"
+	fieldKubernetesCACert  = "kubernetes_ca_cert"
+	fieldServiceAccountJWT = "service_account_jwt"
+	fieldDisableLocalCAJWT = "disable_local_ca_jwt"
+)
+
+func kubernetesSecretBackendResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: kubernetesSecretBackendCreateUpdate,
+		ReadContext:   kubernetesSecretBackendRead,
+		UpdateContext: kubernetesSecretBackendCreateUpdate,
+		DeleteContext: kubernetesSecretBackendDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: getKubernetesSecretsMountSchema(),
+	}
+}
+
+func getKubernetesSecretsMountSchema() map[string]*schema.Schema {
+	s := getMountSchema("type")
+	for k, v := range getKubernetesSecretsSchema() {
+		s[k] = v
+	}
+	return s
+}
+
+func getKubernetesSecretsSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		fieldKubernetesHost: {
+			Type:        schema.TypeString,
+			Description: "The Kubernetes API URL to connect to.",
+			Optional:    true,
+			Computed:    true,
+		},
+		fieldKubernetesCACert: {
+			Type: schema.TypeString,
+			Description: "A PEM-encoded CA certificate used by the secret engine to " +
+				"verify the Kubernetes API server certificate. Defaults to the " +
+				"local pod’s CA if found, or otherwise the host's root CA set.",
+			Optional: true,
+			Computed: true,
+		},
+		fieldServiceAccountJWT: {
+			Type: schema.TypeString,
+			Description: "The JSON web token of the service account used by the " +
+				"secrets engine to manage Kubernetes credentials. Defaults to the " +
+				"local pod’s JWT if found.",
+			Optional:  true,
+			Sensitive: true,
+		},
+		fieldDisableLocalCAJWT: {
+			Type: schema.TypeBool,
+			Description: "Disable defaulting to the local CA certificate and service " +
+				"account JWT when running in a Kubernetes pod.",
+			Optional: true,
+			Default:  false,
+		},
+	}
+}
+
+func kubernetesSecretBackendCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var path string
+	if d.IsNewResource() {
+		path = d.Get(consts.FieldPath).(string)
+		if err := createMount(d, client, path, consts.MountTypeKubernetes); err != nil {
+			return diag.FromErr(err)
+		}
+	} else {
+		if err := updateMount(d, meta, true); err != nil {
+			return diag.FromErr(err)
+		}
+		path = d.Id()
+	}
+	d.SetId(path)
+
+	data := make(map[string]interface{})
+	fields := []string{
+		fieldKubernetesCACert,
+		fieldServiceAccountJWT,
+		fieldDisableLocalCAJWT,
+	}
+	for _, k := range fields {
+		if d.HasChange(k) {
+			data[k] = d.Get(k)
+		}
+	}
+
+	// kubernetes_host always needs to be provided on configuration updates.
+	// Otherwise, an error will occur if the KUBERNETES_SERVICE_HOST and
```

Comment:
This would be in the context of a k8s  pod?

Reply:
If Vault isn't running in k8s, then `kubernetes_host` always needs to be provided (even if it's just the empty string). Otherwise, we'll get an error when we try to make the update. You can see this from the logic at https://github.com/hashicorp/vault-plugin-secrets-kubernetes/blob/main/path_config.go#L126-L130.